stm32h745i_disco/stm32h750b_dk: fix instabilites in SDRAM accesses, add full SDRAM access test

etienne-lms opened the pull request:
Propose a FMC/SDRAM interface configuration change since some instabilites were found when using **tests/drivers/memc** to test more access to these boards SDRAM than the 64 first bytes.

Correct SDRAM size on stm32f746g_disco and stm32f7508_dk boards since their user manual specifies that only half of the SDRAM physical capacity is accessible.

This P-R proposes to enable full SDRAM write/read accesses for these ST boards that embed an SDRAM through an FMC interface.

This change is based on 2 commits already under review in P-R https://github.com/zephyrproject-rtos/zephyr/pull/99799.
(now merged)

etienne-lms committed:
boards: st: stm32h750b_dk: increase FMC SDRAM clock period

Testing full SDRAM access on stm32h750b_disco showed instabilities
and corrupted accessed. Increasing the FMC SDRAM clock period fixes
the issue.

This change ensures stability of transactions with the SDRAM but
may be sub-optimized regarding performances. An alternate correction
would need further investigations in the FMC interface timings
and clocks configuration.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
tests: drivers: memc: fix minor inconsistencies in offsets

Correct memc driver test application to allocated only the required
size for the tests and print valid information instead of byte
offset that was actually a 32bit cell pointer offset.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
tests: drivers: memc: test full SDRAM on a few stm32 boards

Add an overlay for some STM32 boards that embed an SDRAM connected
through FMC interface to better cover their accessibility. Despite
each SDRAM is a few MByte large test are not too long lasting:
- stm32f429i_disc1 .... 8MByte: <1s
- stm32f469i_disco ... 16MByte: <1.5s
- stm32f746g_disco .... 8MByte: <1s
- stm32f7508_dk ......  8MByte: not tested
- stm32f769i_disco ... 16MByte: ~2s
- stm32h745i_disco .... 8MByte: <0.2s
- stm32h747i_disco ... 32MByte: <0.6s
- stm32h750b_dk ....... 8Mbyte: <0.2s
- stm32h757i_eval .... 32MByte: not tested
- stm32h7b31i_dk ..... 16MByte: not tested

Adding these test ensures non-regression on SDRAM support.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
boards: st: stm32h745i_disco: increase FMC SDRAM clock period

Testing full SDRAM access on stm32h745i_disco showed instabilities
and corrupted accessed. Increasing the FMC SDRAM clock period fixes
the issue.

This change ensures stability of transactions with the SDRAM but
may be sub-optimized regarding performances. An alternate correction
would need further investigations in the FMC interface timings
and clocks configuration.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

djiatsaf-st reviewed (dismissed):
No concerns regarding the test bench about the test duration reported in your last commit.

LGTM

erwango commented:
`DNM` could be removed once https://github.com/zephyrproject-rtos/zephyr/pull/99799 is merged

etienne-lms committed:
boards: st: stm32f746g_disco: Fix sdram available size

According to UM1907 Rev 6 section 5.13 "SDRAM Memory" [1], the
available SDRAM size is only 8MByte since only the lowest 64Mbit are
accessible despite the SDRAM capacity is 16MByte (128MBit).

Link: https://www.st.com/resource/en/user_manual/um1907-discovery-kit-for-stm32f7-series-with-stm32f746ng-mcu-stmicroelectronics.pdf [1]
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
boards: st: stm32f7508_dk: Fix sdram available size

According to UM2470 Rev 2 section 4.13 "SDRAM memory" [1], the
available SDRAM size is only 8MB since only the lowest 64-Mbit
are accessible despite the SDRAM capacity is 16MByte (128MBit).

Link: https://www.st.com/resource/en/user_manual/um2470-discovery-kit-for-stm32f7-series-with-stm32f750n8-mcu-stmicroelectronics.pdf [1]
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented:
Rebased since #99799 has been merged.

I've added 2 commits to fix 2 board SDRAM sizes and enabled full SDRAM write/read tests (memc test) and some more ST boards to cover all those embedding an SDRAM connected through an FMC interface.
